packages/wonder-blocks-modal/components/unique-dialog.js

```diff
@@ -0,0 +1,63 @@
+// @flow
+import * as React from "react";
+
+import {UniqueIDProvider} from "@khanacademy/wonder-blocks-core";
+import type {IIdentifierFactory} from "@khanacademy/wonder-blocks-core";
+
+type Props = {|
+    /**
+     * Use the children-as-function pattern to pass a uniqueId string for
+     * use anywhere within children. This provides a way of adding a unique title
+     * to an existant modal variation (OnePaneDialog) or a custom modal implementation.
+     */
+    children: (uniqueId?: string) => React.Node,
+
+    /**
+     * An optional id parameter for the title. If one is
+     * not provided, a unique id will be generated.
+     */
+    id?: string,
+
+    /**
+     * Test ID used for e2e testing.
+     */
+    testId?: string,
+|};
+
+/**
+ * This is a Dialog wrapper that returns a unique Modal Title for A11Y purposes.
+ * This way, the wrapped Dialog will receive this custom ID and will use it to refer to a
+ * visible dialog title.
+ *
+ * @see https://www.w3.org/TR/wai-aria-practices/#dialog_roles_states_props
+ */
+export default class UniqueDialog extends React.Component<Props> {
```

Comment:
Added this functionality in a new component for reusability/composition purposes. This way, if a dev needs to create his own Modal/Dialog variation (e.g. TwoPanelDialog), he can rely on this component to add this unique title to the dialog.

e.g.
```
<UniqueDialog>
    {(id) => <CustomTwoPaneDialog id={id}> .... </CustomTwoPaneDialog>}
</UniqueDialog>
```

Reply:
I wonder if we can make a more generic variation of this component. I mean, there's nothing `modal`-specific here except the scope. Perhaps we can elevate scope to props and then make this be a drop in replacement for `UniqueIDProvider`?

Reply:
That makes total sense!. This task will be implemented here: https://khanacademy.atlassian.net/browse/WB-576